File: ROS2_WS/src/my_cpp_package/src/number_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"

class NumberPublisherNode : public rclcpp::Node{
    public:
        NumberPublisherNode() : Node("Number_Publisher_CPP"){
            this->declare_parameter("number_to_publish", 10);
            this->declare_parameter("publish_frequency", 1.0);

            number_ = this->get_parameter("number_to_publish").as_int();
            double frequency_ = this->get_parameter("publish_frequency").as_double();
            number_publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number", 10);
            number_timer_ = this->create_wall_timer(std::chrono::milliseconds((int) (1000.0 / frequency_)), std::bind(&NumberPublisherNode::publishNumber, this));
            RCLCPP_INFO(this->get_logger(), "CPP Number Publisher Node has been started");
        }
    private:
        void publishNumber(){
            auto msg = example_interfaces::msg::Int64();
            msg.data = number_;
            number_publisher_->publish(msg);
        }

    int number_;
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr number_publisher_;
    rclcpp::TimerBase::SharedPtr number_timer_;
};

int main(int argc, char **argv){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NumberPublisherNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: ROS2_WS/src/my_python_pkg/my_python_pkg/number_publisher.py
import rclpy
from rclpy.node import Node

from example_interfaces.msg import Int64

class NumberPublisher(Node):
    def __init__(self):
        super().__init__("number_publisher")
        self.declare_parameter("number_to_publish", 15)
        self.declare_parameter("publish_frequency", 1.0)

        self.publish_frequency_ = self.get_parameter("publish_frequency").value
        self.number_ = self.get_parameter("number_to_publish").value
        self.number_publisher_ = self.create_publisher(Int64, "number", 10)
        self.number_timer_ = self.create_timer(1.0 / self.publish_frequency_, self.publish_number)
        self.get_logger().info("Number publisher node has been started")

    def publish_number(self):
        msg = Int64()
        msg.data = self.number_
        self.number_publisher_.publish(msg)


def main(args=None):
    rclpy.init(args=args)
    node = NumberPublisher()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__mian__":
    main()